tests/providers/google/marketing_platform/hooks/test_analytics.py

```diff
@@ -30,24 +34,31 @@
 GCP_CONN_ID = "test_gcp_conn_id"
 DELEGATE_TO = "TEST_DELEGATE_TO"
 IMPERSONATION_CHAIN = ["ACCOUNT_1", "ACCOUNT_2", "ACCOUNT_3"]
+DEPRECATION_MESSAGE = (
+    r"Call to deprecated class GoogleAnalyticsHook\."
+    r" \(The `GoogleAnalyticsHook` class is deprecated,"
+    r" please use `GoogleAnalyticsAdminHook` instead\.\)"
+)
 
 
 class TestGoogleAnalyticsHook:
     def setup_method(self):
         with mock.patch(
             "airflow.providers.google.common.hooks.base_google.GoogleBaseHook.__init__",
             new=mock_base_gcp_hook_default_project_id,
-        ):
+        ), warnings.catch_warnings():
+            warnings.simplefilter("ignore", AirflowProviderDeprecationWarning)
             self.hook = GoogleAnalyticsHook(API_VERSION, GCP_CONN_ID)
 
     @mock.patch("airflow.providers.google.common.hooks.base_google.GoogleBaseHook.__init__")
     def test_init(self, mock_base_init):
-        hook = GoogleAnalyticsHook(
-            API_VERSION,
-            GCP_CONN_ID,
-            delegate_to=DELEGATE_TO,
-            impersonation_chain=IMPERSONATION_CHAIN,
-        )
+        with pytest.warns(AirflowProviderDeprecationWarning, match=DEPRECATION_MESSAGE):
+            hook = GoogleAnalyticsHook(
+                API_VERSION,
+                GCP_CONN_ID,
+                delegate_to=DELEGATE_TO,
+                impersonation_chain=IMPERSONATION_CHAIN,
+            )
```

Comment:
I am not sure if this is right?
The goal of `TestGoogleAnalyticsHook` is to test the hook so I think this should work directly with `GoogleAnalyticsAdminHook` and we should have a test that make sure if we use `GoogleAnalyticsHook` it raises deprecation warning.

WDYT?

Reply:
I believe there is already another class `TestGoogleAnalyticsAdminHook` which tests the `GoogleAnalyticsAdminHook`:
https://github.com/apache/airflow/blob/2149b4dbee8fb524bb235280aaef158afaec8d4a/tests/providers/google/marketing_platform/hooks/test_analytics_admin.py#L43-L47

`TestGoogleAnalyticsHook` will not be needed once `GoogleAnalyticsHook` is deprecated, as the new hook already has its own tests under `TestGoogleAnalyticsAdminHook`. 

Just adding `pytest.warns` (keeping the current imp) to `TestGoogleAnalyticsHook` seems appropriate to me.

WDYT?

Reply:
Makes sense.